pkg/cloud/talos/provisioners/packet/packet.go

```diff
@@ -0,0 +1,227 @@
+package packet
+
+import (
+	"errors"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/packethost/packngo"
+	"github.com/talos-systems/cluster-api-provider-talos/pkg/cloud/talos/utils"
+	"gopkg.in/yaml.v2"
+	"k8s.io/client-go/kubernetes"
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+// Packet represents a provider for Packet.
+type Packet struct {
+	client *packngo.Client
+}
+
+// ClusterInfo holds data about desired config in cluster object
+type ClusterInfo struct {
+	ProjectID string
+	Instances InstanceInfo
+}
+
+// InstanceInfo holds data about the instances we'll create
+type InstanceInfo struct {
+	Plan     string
+	Facility string
+	PXEUrl   string
+	Install  map[string]interface{}
+}
+
+// Userdata holds userdata in struct form
+type Userdata struct {
+	Version    string
+	Security   interface{}
+	Services   interface{}
+	Install    map[string]interface{}
+	Networking Network
+}
+
+type Network struct {
+	OS OS
+}
+
+type OS struct {
+	Devices []map[string]interface{}
+}
+
+//NewPacket returns an instance of the Packet provisioner
+func NewPacket() (*Packet, error) {
+	c, err := packngo.NewClient()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return &Packet{client: c}, nil
+}
+
+// Create creates an instance in Packet.
+func (packet *Packet) Create(cluster *clusterv1.Cluster, machine *clusterv1.Machine, clientset *kubernetes.Clientset) error {
+
+	clusterSpec, err := utils.ClusterProviderFromSpec(cluster.Spec.ProviderSpec)
+	if err != nil {
+		return err
+	}
+
+	machineSpec, err := utils.MachineProviderFromSpec(machine.Spec.ProviderSpec)
+	if err != nil {
+		return err
+	}
+
+	packetConfig := &ClusterInfo{}
+	yaml.Unmarshal([]byte(machineSpec.Platform.Config), packetConfig)
+
+	// Here we pull down the userdata config map and add the install section to the end if it's defined in the machine
+	udConfigMap, err := utils.FetchConfigMap(cluster, machine, clientset)
+	if err != nil {
+		return err
+	}
+	udStruct := &Userdata{}
+	yaml.Unmarshal([]byte(udConfigMap.Data["userdata"]), udStruct)
+
+	if packetConfig.Instances.Install != nil {
+		udStruct.Install = packetConfig.Instances.Install
+	}
+
+	//Add network tweaks for elastic IPs to userdata
+	floatingIP := ""
```

Comment:
I think they're both equivalent, but typically we'd do 
```
var floatingIP string
```
in this case.

Reply:
Fixed